packages/legacy/core/App/modules/openid/context/OpenIDCredentialRecordProvider.tsx

```diff
@@ -0,0 +1,151 @@
+  const newRecordsState = [...state.w3cCredentialRecords]
+  const index = newRecordsState.findIndex((r) => r.id === record.id)
+  if (index > -1) {
+    newRecordsState.splice(index, 1)
+  }
+
+  return {
+    ...state,
+    w3cCredentialRecords: newRecordsState,
+  }
+}
+
+const defaultSate: OpenIDCredentialRecordState = {
+  openIDCredentialRecords: [],
+  w3cCredentialRecords: [],
+  sdJwtVcRecords: [],
+  isLoading: true,
+}
+
+interface Props {
+  children: React.ReactNode
+}
+
+const OpenIDCredentialRecordContext = createContext<OpenIDCredentialContext>(null as unknown as OpenIDCredentialContext)
+
+const isW3CCredentialRecord = (record: W3cCredentialRecord) => {
+  return record.getTags()?.claimFormat === ClaimFormat.JwtVc
+}
+
+const filterW3CCredentialsOnly = (credentials: W3cCredentialRecord[]) => {
+  return credentials.filter((r) => isW3CCredentialRecord(r))
+}
+
+// eslint-disable-next-line react/prop-types
+export const OpenIDCredentialRecordProvider: React.FC<PropsWithChildren<Props>> = ({ children }) => {
+  const [state, setState] = useState<OpenIDCredentialRecordState>(defaultSate)
+
+  const { agent } = useAgent()
+  const [logger] = useServices([TOKENS.UTIL_LOGGER])
+
+  function checkAgent() {
+    if (!agent) {
+      const error = 'Agent undefined!'
+      logger.error(`[OpenIDCredentialRecordProvider][deleteCredential] ${error}`)
+      throw new Error(error)
+    }
+  }
+
+  async function storeCredential(cred: W3cCredentialRecord | SdJwtVcRecord): Promise<void> {
+    checkAgent()
+    if (cred instanceof W3cCredentialRecord) {
+      await agent?.dependencyManager.resolve(W3cCredentialRepository).save(agent.context, cred)
+    } else {
+      await agent?.dependencyManager.resolve(SdJwtVcRepository).save(agent.context, cred)
+    }
+  }
+
+  async function deleteCredential(cred: W3cCredentialRecord | SdJwtVcRecord) {
+    checkAgent()
+    if (cred instanceof W3cCredentialRecord) {
+      await agent?.w3cCredentials.removeCredentialRecord(cred.id)
+    } else if (cred instanceof SdJwtVcRecord) {
+      await agent?.sdJwtVc.deleteById(cred.id)
+    }
+  }
+
+  useEffect(() => {
+    if (!agent) {
+      return
+    }
+    agent.w3cCredentials?.getAllCredentialRecords().then((w3cCredentialRecords) => {
+      setState({ ...state, w3cCredentialRecords: filterW3CCredentialsOnly(w3cCredentialRecords), isLoading: false })
+    })
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [agent])
+
+  useEffect(() => {
+    if (!state.isLoading && agent) {
+      const credentialAdded$ = recordsAddedByType(agent, W3cCredentialRecord).subscribe((record) => {
+        //This handler will return ANY creds added to the wallet even DidComm
+        //Sounds like a bug in the hooks package
+        //This check will safe guard the flow untill a fix goes to the hooks
+        if (isW3CCredentialRecord(record)) {
+          setState(addRecord(record, state))
+        }
```

Comment:
I thought all credentials in Credo agent get converted to `W3cCredentialRecord` type after migrations are done? I have asked the question in the Discord but it's worth double checking that cause a lot of the logic here depends on checking that `W3cCredentialRecord` type

Reply:
This is not the case I believe now, it's just a bug in the hooks package and specifically in the handler. Records in wallet are not mixing AnonCreds with W3c. and you can notice that from the check that I made on the Connection screen